tests/operators/test_bash.py

```diff
@@ -123,6 +125,36 @@ def test_command_not_found(self):
         ):
             BashOperator(task_id='abc', bash_command='set -e; something-that-isnt-on-path').execute({})
 
+    def test_command_without_cwd(self):
+        val = "xxxx"
+        op = BashOperator(task_id='abc', bash_command=f'set -e; echo "{val}";')
+        line = op.execute({})
+        assert line == val
+
+    def test_command_with_cwd(self):
+
+        # The command is echo a string to a file into cwd folder and return the string as the output
+        test_cmd = f'set -e; echo "xxxx" |tee outputs.txt'
+
+        with TemporaryDirectory(prefix='test_command_with_cwd') as test_cwd_folder:
+            # Test if the cwd is a file_path
+            test_1_file_path = f'{test_cwd_folder}/test1.txt'
+            with open(test_1_file_path, 'w') as tmp_file:
+                tmp_file.write("xxxxx")
+                with pytest.raises(AirflowException, match=f"The cwd {test_1_file_path} must be a directory"):
+                    BashOperator(
+                        task_id='abc', bash_command=test_cmd, cwd=f'{test_cwd_folder}/test1.txt'
+                    ).execute({})
```

Comment:
This should be split into a separate function (and use `TemporaryFile` instead).

It’s probably best to split `test_command_with_cwd` into three functions:

1. `cwd` does not exist.
2. `cwd` exists but is not a directory.
3. `cwd` points to a valid directory.

Reply:
Thanks for the suggestions.
Changed it already.